File: app/src/main/java/com/bkjcb/rqapplication/adapter/GasWorkRecordAdapter.java
package com.bkjcb.rqapplication.adapter;

import android.text.TextUtils;

import com.bkjcb.rqapplication.MyApplication;
import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.model.GasUserRecordResult;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

/**
 * Created by DengShuai on 2020/5/6.
 * Description :
 */
public class GasWorkRecordAdapter extends BaseQuickAdapter<GasUserRecordResult.GasUserRecord, BaseViewHolder> {

    public GasWorkRecordAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, GasUserRecordResult.GasUserRecord item) {
        helper.setText(R.id.check_time, "建档日期：" + item.getJiandangriqi())
                .setText(R.id.check_name, item.getYonghuming())
                .setText(R.id.check_operate, item.getDizhi())
                .setBackgroundColor(R.id.check_divider, getColor(item.getRquserid()))
                .addOnClickListener(R.id.check_type);
    }

    private String getStatus(String status) {
        if (TextUtils.isEmpty(status)) {
            return "#未关联";
        }
        return "#已关联";
    }

    private int getColor(String type) {
        if (!TextUtils.isEmpty(type)) {
            return getColor(R.color.colorMint);
        } else {
            return getColor(R.color.colorTextThird);
        }
    }

    private int getColor(int resID) {
        return MyApplication.getContext().getResources().getColor(resID);
    }

    private int getColors(String type) {
        if (type.equals("0")) {
            return getColor(R.color.colorMint);
        }
        return getColor(R.color.colorAccent);
    }

}

File: app/src/main/java/com/bkjcb/rqapplication/gasrecord/adapter/GasWorkRecordAdapter.kt
package com.bkjcb.rqapplication.gasrecord.adapter

import android.text.TextUtils
import com.bkjcb.rqapplication.MyApplication
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.gasrecord.model.GasUserRecordResult.GasUserRecord
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder

/**
 * Created by DengShuai on 2020/5/6.
 * Description :
 */
class GasWorkRecordAdapter(layoutResId: Int) : BaseQuickAdapter<GasUserRecord, BaseViewHolder>(layoutResId) {
    override fun convert(helper: BaseViewHolder, item: GasUserRecord) {
        helper.setText(R.id.check_time, "建档日期：" + item.jiandangriqi)
                .setText(R.id.check_name, item.yonghuming)
                .setText(R.id.check_operate, item.dizhi)
                .setBackgroundColor(R.id.check_divider, getColor(item.rquserid))
                .addOnClickListener(R.id.check_type)
    }

    private fun getColor(type: String): Int {
        return if (!TextUtils.isEmpty(type)) {
            getColor(R.color.colorMint)
        } else {
            getColor(R.color.colorTextThird)
        }
    }

    private fun getColor(resID: Int): Int {
        return MyApplication.getContext().resources.getColor(resID)
    }

    private fun getColors(type: String): Int {
        return if (type == "0") {
            getColor(R.color.colorMint)
        } else getColor(R.color.colorAccent)
    }
}